### DISCUSSION TITLE
Error Compiling v2.3.3 under Ubuntu

### DISCUSSION THREAD

**Discussion by rebeltaz** [audacity/audacity#4160] [Q&A]
I am trying to compile Audacity v2.3.3 under Ubuntu 22.04. I compiled and installed wxWidgets as instructed. I already have the required files installed. Compiling went fine. When I run make, though, I a getting two errors. I have tried creating an account on the adacity forum, but I am not getting the activation emails, so hopefully someone here can help?

This is the error I am getting:

```
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:2594: audacity] Error 1
make[2]: Leaving directory '/home/derek/audacity-Audacity-2.3.3/build/src'
make[1]: *** [Makefile:1749: all] Error 2
make[1]: Leaving directory '/home/derek/audacity-Audacity-2.3.3/build/src'
make: *** [Makefile:811: all-recursive] Error 1

```

**Comment by crsib** [comment]
The log is quite a bit incomplete. I would like to mention, that Audacity 2.x is not longer supported and it is not guaranteed that it builds correctly on modern Linux distributions

**Reply by rebeltaz** [reply]
> The log is quite a bit incomplete. I would like to mention, that Audacity 2.x is not longer supported and it is not guaranteed that it builds correctly on modern Linux distributions

I'm sorry. Yeah. I just grabbed that from the terminal screen. The scroll wasn't big enough to get all of it. If you can tell me how to get a verbose log, I can post all of it. I understand there are no guarantees.

**Comment by rebeltaz** [comment]
So I figured out how to retain the log and configure was giving a warning for a missing library - libsndfile - so I installed libsndfile1-dev and that cleared that warning. Now it's complaining about missing libflac, libogg, and libvorbis libraries. I checked and the -dev libraries are installed - 

libflac-dev:amd64                             1.3.3-2ubuntu0.1
libogg-dev:amd64                              1.3.5-0ubuntu3
libvorbis-dev:amd64                           1.3.7-1build2

These are the only -dev files I see available for these libraries:

```
derek@laptop:~/audacity-Audacity-2.3.3/build$ apt-cache search libvorbis | grep dev
libvorbis-dev - development files for Vorbis General Audio Compression Codec
libvorbis-ocaml-dev - OCaml bindings for the vorbis library
libvorbisidec-dev - Integer-only Ogg Vorbis decoder, AKA "tremor" (Development Files)

derek@laptop:~/audacity-Audacity-2.3.3/build$ apt-cache search libogg | grep dev
libogg-dev - Ogg bitstream library development files
libfishsound1-dev - simple API that wraps Xiph.Org audio codecs (development files)
libogg-ocaml-dev - OCaml bindings for the Ogg bitstream library
liboggkate-dev - Codec for karaoke and text encapsulation for Ogg (dev)
liboggplay1-dev - Library for playing OGG multimedia (development files)
liboggz2-dev - convenience interface for Ogg stream I/O (development files)
libopusfile-dev - High-level API for basic manipulation of Ogg Opus audio streams

derek@laptop:~/audacity-Audacity-2.3.3/build$ apt-cache search libflac | grep dev
libflac++-dev - Free Lossless Audio Codec - C++ development library
libflac-dev - Free Lossless Audio Codec - C development library
libflac-ocaml-dev - OCaml interface to the flac library -- development files
```
I assume then that the version is the problem?

[configure_log.txt](https://github.com/audacity/audacity/files/10416321/configure_log.txt)

**Reply by rebeltaz** [reply]
still looking for an solution if anyone has any insight...

**Reply by crsib** [reply]
I wonder why libsndfile attempts to be built from the Audacity source tree. A system version should be used, as you have installed it. 

I am not familiar with the make based Audacity build system at all, so I'm afraid I won't be able to help.

**Reply by rebeltaz** [reply]
> I wonder why libsndfile attempts to be built from the Audacity source tree. A system version should be used, as you have installed it.
> 
> I am not familiar with the make based Audacity build system at all, so I'm afraid I won't be able to help.

Because you mentioned that, I played around with it again tonight and I was actually able to get past that. I swapped 'local' and 'system' in the recommended build command - 

../configure --with-lib-preference="local system" 

instead of 

../configure --with-lib-preference="system local"

I got a lot further in the 'make' process, but now it's hanging on:

```
/usr/bin/ld: ../lib-src/libnyquist/libnyquist.a(midifns.o):(.bss+0x0): multiple definition of `IOinputfd'; ../lib-src/libnyquist/libnyquist.a(userio.o):(.bss+0x0): first defined here
/usr/bin/ld: ../lib-src/libnyquist/libnyquist.a(midifns.o):(.bss+0x4): multiple definition of `IOnochar'; ../lib-src/libnyquist/libnyquist.a(userio.o):(.bss+0x4): first defined here
```

I'm going to keep trying, but if you or anyone else knows about this... I'm still open. Thanks again.

[make_log.txt](https://github.com/audacity/audacity/files/10473756/make_log.txt)
